data/mods/partnersincrime/scripts.ts

```diff
@@ -0,0 +1,341 @@
+export const Scripts: ModdedBattleScriptsData = {
+	gen: 8,
+	inherit: 'gen8',
+	nextTurn() {
+		this.turn++;
+		this.lastSuccessfulMoveThisTurn = null;
+
+		// Partners in Crime moveSlot updating
+		// Must be highest priority so imprison doesn't lag behind.
+		for (const side of this.sides) {
+			for (const pokemon of side.active) {
+				pokemon.moveSlots = pokemon.moveSlots.filter(move => move.originalPoke === pokemon.m.value);
+				const ally = side.active.find(mon => mon && mon !== pokemon && !mon.fainted);
+				let allyMoves = ally ? this.dex.deepClone(ally.moveSlots) : [];
+				if (ally) {
+					// @ts-ignore
```

Comment:
i don't like using ts-ignore here, it's better to typecast as `any` typically. What was the error originally?

Reply:
this can honestly be merged with the above line but I want to find a solution to transform that doesn't bleed into pokemon.ts first before i handle this.

Reply:
you can keep it, just change the typecast

Reply:
ig keep this for now, try to figure out an alternative to it in the meantime